### ISSUE TITLE
Should `exposes` terms be sorted?

### ISSUE THREAD

**Issue by rtfeldman** [avh4/elm-format#319]
In https://github.com/avh4/elm-format/pull/279#issuecomment-263102133 @avh4 asked:

> should exposings in import lines be sorted?
> should exposings in the module line be sorted? (This is what I've referred to in some of my videos)

This is a separate discussion from https://github.com/avh4/elm-format/issues/8 (sorting the imports themselves by module name and/or alias).

## Explain the Benefits

If you have a preference, please **explain the benefits** of doing things your way, not just "this is what I do." (Everyone has a way they do it, and it's not possible to figure out what's best without understanding the benefits and drawbacks of different alternatives!)

**Comment by stil4m** [comment]
I'm not sure about this. Normally I sort these by the order the functions occur in my file. Sometimes I deviate from this, putting the all the types and aliases first.

Would sorting be alphabetical or should this be a different kind of sorting algorithm?

**Comment by basile-henry** [comment]
I think sorting the exposed variables in the module header would be a good thing to do. The same reasoning as for the discussion in #8 applies: consistency, better to read and share code with people, possibly fewer merged conflicts.

I have just added a new commit to #279 to handle this so we can see how it would behave more easily. The current sorting is done alphabetically but by kind of variables. First the types are exposed, then the operators and finally the functions.

For example:

```elm
module Test exposing ((|>), a, B, c, A, b)
```

is formatted:

```elm
module Test exposing (A, B, (|>), a, b, c)
```

**Comment by iuriikomarov** [comment]
Split into two blocks: external packages and internal imports. Empty line between. Sort alphabetically inside.

**Comment by lydell** [comment]
I usually think about the order any time I add something new to import (can’t help it). Then I never look at those imports again. That’s me spending time thinking about pointless details instead of about how my program should work. So if elm-format would sort them for me it would be a time saver, I think.

**Comment by avh4** [comment]
This is implemented in cf5d642f...44296126, which also merges any duplicates that are found.

The current sort order is: infix operators, data types, other values/functions. (and alphabetical or ASCII order within each group)
